presto-hive/src/main/java/com/facebook/presto/hive/metastore/thrift/ThriftMetastoreUtil.java

```diff
@@ -586,4 +608,29 @@ private static OptionalLong toMetastoreDistinctValuesCount(OptionalLong distinct
         }
         return distinctValuesCount;
     }
+
+    public static Set<ColumnStatisticType> getSupportedColumnStatistics(Type type)
+    {
+        if (type.equals(BOOLEAN)) {
+            return ImmutableSet.of(NUMBER_OF_NON_NULL_VALUES, NUMBER_OF_TRUE_VALUES);
+        }
+        if (isNumericType(type) || type.equals(DATE) || type.equals(TIMESTAMP)) {
+            return ImmutableSet.of(NUMBER_OF_NON_NULL_VALUES, MIN, MAX, NUMBER_OF_DISTINCT_VALUES);
+        }
+        if (isVarcharType(type)) {
+            return ImmutableSet.of(NUMBER_OF_NON_NULL_VALUES, NUMBER_OF_DISTINCT_VALUES, MAX_VALUE_SIZE_IN_BYTES, AVERAGE_VALUE_SIZE_IN_BYTES);
+        }
+        if (isCharType(type)) {
+            return ImmutableSet.of(NUMBER_OF_NON_NULL_VALUES, NUMBER_OF_DISTINCT_VALUES);
```

Comment:
> Isn't AVERAGE_VALUE_SIZE_IN_BYTES interesting?

It is not obvious how to compute it for CHAR. Probably some custom function will be needed. I decided to skip it for now. In the optimizer we can use the length from the type itself. Usually the deviation is not major (or otherwise there is less sense of using CHAR)

Reply:
a TODO note?

Reply:
Don't forget to add this TODO about `*_VALUE_SIZE_IN_BYTES`

Reply:
> Don't forget to add this TODO about *_VALUE_SIZE_IN_BYTES

I'm going to introduce it in a very next PR